.github/workflows/quarkus-nightly.yml

```diff
@@ -0,0 +1,78 @@
+name: Quarkus Nightly
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - quarkus-nightly
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: quarkus-nightly-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  rebase-quarkus-nightly-branch:
+    name: Rebase quarkus-nightly branch
+    if: github.event_name != 'schedule' || github.repository == 'keycloak/keycloak'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: quarkus-nightly
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@github.com"
+
+      - name: Fetch and rebase
+        run: |
+          git fetch origin quarkus-nightly
+          git rebase origin/main
+          if [ "$(grep -oP '(?<=<quarkus.version>).*?(?=<\/quarkus.version>)' ${GITHUB_WORKSPACE}/pom.xml)" != "999-SNAPSHOT" ]; then
+            echo "Error: <quarkus.version> in pom.xml is not set to 999-SNAPSHOT."
+            exit 1
+          fi
+
+      - name: Push changes
+        run: |
+          git push -f origin HEAD:quarkus-nightly
+
+  run-nightly-quarkus:
+    name: Run workflows with the nightly Quarkus release
+    runs-on: ubuntu-latest
+    needs: rebase-quarkus-nightly-branch
```

Comment:
We should be able to run the tests regardless if rebase was skipped or not. I think this prevents it, or am I wrong?

Reply:
Yes, it prevents it - all the jobs are executed sequentially, see e.g.: https://github.com/Pepo48/keycloak/actions/runs/6581639054

Initially, I designed it this way on purpose - my understanding was that the successful rebase should be mandatory. If it fails due to a conflict, we won't run the tests.